extensions/resource-deployment/src/services/tools/kubeCtlTool.ts

```diff
@@ -42,8 +49,9 @@ export class KubeCtlTool extends ToolBase {
 	protected getVersionFromOutput(output: string): SemVer | undefined {
 		let version: SemVer | undefined = undefined;
 		if (output) {
-			const versionJson = JSON.parse(output);
-			version = new SemVer(`${versionJson.clientVersion.major}.${versionJson.clientVersion.minor}.0`);
+			const versionJson: KubeCtlVersion = JSON.parse(output);
+			// kubectl version output might contain '+' character in the minor version, e.g. 16+, we have to remove it to make it a valid semantic version string.
+			version = new SemVer(`${versionJson.clientVersion.major}.${versionJson.clientVersion.minor.replace(/\+/g, '')}.0`);
```

Comment:
Semver should be able to handle that version string from the Beta. I think the issue was that we were constructing a version string that was stripping parts of the version off. We should be sending  "v1.16.6-beta.0" from the output to SemVer constructor instead of doing our own parsing. So pass versionJson.gitVersion to the SemVer constructor.

Reply:
you are right, unfortunately I've merged the changes, i will have a separate PR.